src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2361,38 +2427,38 @@ ManageWorkerPool(WorkerPool *workerPool)
 		 * Number of additional connections we would need to run all ready tasks in
 		 * parallel.
 		 */
-		int newConnectionsForReadyTasks = readyTaskCount - usableConnectionCount;
+		int newConnectionsForReadyTasks = Max(0, readyTaskCount - usableConnectionCount);
```

Comment:
Here there is an additional Max because newConnectionsForReadyTasks cannot be negative. If it is set to zero we won't open any new connection.